tests/contrib/celery/test_integration.py

```diff
@@ -0,0 +1,225 @@
+from nose.tools import eq_, ok_
+
+from .utils import CeleryTestCase
+
+
+class CeleryIntegrationTask(CeleryTestCase):
+    """
+    Ensures that the tracer works properly with a real Celery application
+    without breaking the Application or Task APIs.
+    """
+    def test_concurrent_delays(self):
+        # it should create one trace for each delayed execution
+        @self.app.task
+        def fn_task():
+            return 42
+
+        for x in range(100):
+            fn_task.delay()
+
+        traces = self.tracer.writer.pop_traces()
```

Comment:
Is there a risk that we didn't finish all the tasks by the moment we arrive here? (I don't know celery internals, even if the task is trivial, can they be queued for later?)